common/src/main/java/org/opensearch/ml/common/transport/prompt/MLCreatePromptInput.java

```diff
@@ -0,0 +1,184 @@
+package org.opensearch.ml.common.transport.prompt;
+
+import lombok.Builder;
+import lombok.Data;
+import lombok.Setter;
+import org.opensearch.Version;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.common.io.stream.Writeable;
+import org.opensearch.core.xcontent.ToXContentObject;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.core.xcontent.XContentParser;
+import org.opensearch.ml.common.CommonValue;
+
+import java.io.IOException;
+import java.util.Map;
+
+import static org.opensearch.core.xcontent.XContentParserUtils.ensureExpectedToken;
+import static org.opensearch.ml.common.CommonValue.TENANT_ID_FIELD;
+import static org.opensearch.ml.common.CommonValue.VERSION_2_19_0;
+import static org.opensearch.ml.common.utils.StringUtils.getParameterMap;
+
+@Data
+public class MLCreatePromptInput implements ToXContentObject, Writeable {
+    public static final String PROMPT_NAME_FIELD = "name";
+    public static final String PROMPT_DESCRIPTION_FIELD = "description";
+    public static final String PROMPT_PROMPT_FIELD = "prompt";
+    public static final String PROMPT_TAG_FIELD = "tag";
+
+    public static final String DRY_RUN_FIELD = "dry_run";
+
+    private static final Version MINIMAL_SUPPORTED_VERSION_FOR_CLIENT_CONFIG = CommonValue.VERSION_2_13_0;
+
+    public static final String DRY_RUN_PROMPT_NAME = "dryRunPrompt";
+
+    private String name;
+    private String description;
+    private Map<String, String> prompt;
+    @Setter
+    private String tenantId;
+    private String tag;
+    private boolean dryRun;
+    //private boolean updatePrompt;
+
+    @Builder(toBuilder = true)
+    public MLCreatePromptInput(
+            String name,
+            String description,
+            Map<String, String> prompt,
+            String tag,
+            boolean dryRun,
+            String tenantId
+//            boolean updatePrompt
+    ) {
+//        if (!dryRun && !updatePrompt) {
+//            if (prompt == null) {
+//                throw new IllegalArgumentException("Prompt is null");
+//            }
+//        }
+        if (!dryRun) {
+            if (prompt == null) {
+                throw new IllegalArgumentException("Prompt is null");
+            }
+        }
+        this.name = name;
+        this.description = description;
+        this.prompt = prompt;
+        this.tag = tag;
```

Comment:
Any of these shouldn't be null? I think at least the name shouldn't be null, so it's better to add validation here.